src/app/static/src/js/store/git/actions.ts

```diff
@@ -40,4 +31,28 @@ export const actions: ActionTree<GitState, RunnerRootState> & Record<GitAction,
                 })
         }
     },
+
+    async [GitAction.RefreshGit](context) {
+        context.commit(GitMutation.SetGitRefreshing)
+        await api.get('/git/fetch/')
+            .then(({data}) => {
+                const gitBranches = data.data.map(branch => branch.name);
+                context.commit(GitMutation.SetFetchedGit, gitBranches)
+                determineSelectedBranch(context)
+            })
+    },
+}
+
+function determineSelectedBranch(context: ActionContext<GitState, RunnerRootState>) {
+    const {branches} = context.state
+    let {selectedBranch} = context.state
+    if (branches.length && !branches.some(branch => branch === selectedBranch)) {
+        selectedBranch = branches[0]
+    }
+    if (!branches.length) {
+        selectedBranch = ""
+    }
+    if (selectedBranch !== context.state.selectedBranch) {
+        context.dispatch('SelectBranch', selectedBranch)
+    }
 }
```

Comment:
```suggestion
function determineSelectedBranch(context: ActionContext<GitState, RunnerRootState>) {
    let {selectedBranch, branches} = context.state
    if (branches.length && selectedBranch) {
        if (!branches.some(branch => branch === selectedBranch)) {
            selectedBranch = branches[0]
        }
    } else {
        await context.dispatch('SelectBranch', selectedBranch)
    }
    }
```

Reply:
I think you're right that the logic could be simplified but I don't think this suggestion would work. Did you mean to put a dispatch after line 50?

Reply:
I ran it manually and it worked, which use case would it not cover ?

Reply:
If I'm reading it correctly, the above will only dispatch the selectedBranch as it is. Lines 49 to 51 won't do anything.